src/HotChocolate/Fusion-vnext/src/Fusion.Composition/PreMergeValidation/PreMergeValidationContext.cs

```diff
@@ -0,0 +1,69 @@
+using HotChocolate.Fusion.Logging.Contracts;
+using HotChocolate.Skimmed;
+
+namespace HotChocolate.Fusion.PreMergeValidation;
+
+internal sealed class PreMergeValidationContext(CompositionContext context)
+{
+    public SchemaDefinition[] SchemaDefinitions => context.SchemaDefinitions;
+    public ICompositionLog Log => context.Log;
+    public IEnumerable<OutputTypeInfo> OutputTypeInfo = [];
+
+    public void Initialize()
+    {
+        InitializeOutputTypeInfo();
+    }
+
+    /// <summary>
+    /// Initializes a structure that makes it easier to access combined output types, fields, and
+    /// arguments for validation purposes.
+    /// </summary>
+    private void InitializeOutputTypeInfo()
+    {
+        OutputTypeInfo = SchemaDefinitions
```

Comment:
Does this end up enumerating each time the calling code uses OutputTypeInfo?

Reply:
Well spotted. Fixed.